release-20.2: backupccl: add more backup telemetry

pbardea opened the pull request:
Backport:
  * 1/1 commits from "backupccl: add more backup telemetry" (#54563)
  * 1/1 commits from "backupccl: update default backup telemetry source to \".manual\"" (#54952)

Please see individual PRs for details.

/cc @cockroachdb/release

pbardea committed:
backupccl: add more backup telemetry

This commit adds telemetry in backup for the type of encryption used in
backups, as well as if the new INTO syntax was used. In addition, it
also adds telemetry to report which backup features are being used by
scheduled backups.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/54863)
<!-- Reviewable:end -->

dt commented on pkg/ccl/backupccl/backup_planning.go:
```diff
@@ -886,39 +886,64 @@ func backupPlanHook(
+			// countSource emits a telemetry counter and also adds a ".scheduled"
+			// suffix if the job was created by a schedule.
+			countSource := func(feature string) {
+				telemetry.Count(feature + sourceSuffix)
```
hm. I wonder if we should keep the global counts for all backups -- scheduled or manual -- and have an additional counter for scheduled? otherwise I wonder if it is prone to forgetting to sum the two counters any time we look at this. I guess a question for @mwang1026

mwang1026 replied:
Subtraction is hard in Looker but addition is easy. If we did go with a global counter, I'd want separate ones for manual and scheduled as well. Thinking through, I personally prefer the latter -- a global counter, and then a split one with manual and scheduled separate. That way dashboards won't break as we transition versions.

pbardea replied:
Just chatted with @mwang1026, it sounds like we can search by a prefix. I think it would be worthwhile updating the default prefix to ".manual" so that there is a way to easily filter telemetry by scheduled and non-scheduled backups.

Put out https://github.com/cockroachdb/cockroach/pull/54952 for this change.

pbardea committed:
backupccl: update default backup telemetry source to ".manual"

This commit updates the telemetry for backup to always include the
source (either ".manual" or ".scheduled"). This allows for folks to
search by a prefix to get the telemetry for all backups, and the
specific source can be appened to filter the telemetry between manual
and scheduled backups.

Release note: None

pbardea commented:
Updated with #54952.
